src/YWidgetFactory.cc

```diff
@@ -316,3 +316,19 @@ YWidgetFactory::createCustomStatusItemSelector( YWidget * parent,
     return createItemSelector( parent,
                                false ); // enforceSingleSelection
 }
+
+
+YMenuBar *
+YWidgetFactory::createMenuBar( YWidget * parent )
+{
+    (void) parent;
+
+    // Default implementation returning 0 to give community-maintained UIs
+    // (libyui-gtk) a chance to catch up with development. Remove this and make
+    // it pure virtual when this is implemented there as well.
+
+    yuiError() << "YMenuBar not implemented in this UI" << endl;
+
+    return 0;
```

Comment:
Nothing is created, so I guess the examples do not work yet, right?

Reply:
This is the default implementation that only becomes relevant if the concrete UI does not reimplement this method.

It is reimplemented in the Qt and the NCurses UI, but not (yet) in the Gtk UI. Providing this default implementation was necessary to give the maintainers of the Gtk UI a chance to catch up when they have time to do so; otherwise (with a pure virtual method) it would have resulted in a build error.

Reply:
@shundhammer thanks a lot for answering, but I did not expect it until you are back :) Enjoy your vacations!